pkg/cmd/cli/cmd/env.go

```diff
@@ -234,7 +243,14 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	}
 
 	skipped := 0
-	for _, info := range infos {
+	oldData := make([][]byte, len(infos))
+	for i, info := range infos {
+		old, err := json.Marshal(info.Object)
```

Comment:
does this even give you versioned JSON? don't you have to use info.Codec to do this (and are you sure that will result in a patch that is valid against the client's API version?)